Update with methods example (#267)
a-h/templ: docs/docs/04-core-concepts/01-components.md

This code is unsafe! In code-only components, you're responsible for escaping the HTML content yourself, e.g. with the `templ.EscapeString` function.
:::

> ## Component as method on types

Reviewer: ```suggestion
## Method components
```

> templ Components can be generated as methods on types

Reviewer: ```suggestion
templ components can be returned from methods (functions attached to types).
```
